### PR TITLE
[KOGITO-6047] Introduce KogitoSource as Binding and Sink

### PR DESCRIPTION
**Pull request by ricardozanini**
# Changes

- :broom: Change the `KogitoSource` interface to behave as a Subject Binding instead to wrap the `KogitoRuntime` object owned by the Kogito Operator
- :wastebasket: Completely removed the Kogito Operator dependencies. Now you can deploy any kind of Podspecable object and bind to this source.

/kind enhancement

**Release Note**

```release-note
`KogitoSource` interface has changed to act as a Subject Binding instead of wrapping the `KogitoRuntime` CR. 
Now the Kogito Operator is not a requirement to use the Kogito Eventing anymore. The subject must be a Kogito application deployed via any Podspecable object (Deployment, KService, DeploymentConfig, and so on).

In the next releases, the source will enforce the subject to be a Kogito service and will add more capabilities and configuration to the service to enhance the integration with the Knative Platform.
```

### CODE REVIEW

File: pkg/reconciler/source/controller.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// NewController for the KogitoSource Bindable interface
+// See: https://github.com/knative/pkg/tree/main/webhook/psbinding for reference
+func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	logger := logging.FromContext(ctx)
+
+	kogitoInformer := kogitosourceinformer.Get(ctx)
+	dc := dynamicclient.Get(ctx)
+	psInformerFactory := podspecable.Get(ctx)
+
+	c := &psbinding.BaseReconciler{
```
Comment (lionelvillard) [inline pkg/reconciler/source/controller.go:34]: I don't see `NamespaceLister` being initialized.

**Comment by ricardozanini** [comment]
@lionelvillard can you take a look? :)

**Review by sutaakar — APPROVED** [review]
Looks good to me

**Pull request merged**